bindings/rust/s2n-tls/src/error.rs

```diff
@@ -8,7 +8,7 @@ use s2n_tls_sys::*;
 use std::{convert::TryFrom, ffi::CStr};
 
 #[non_exhaustive]
-#[derive(Debug, PartialEq)]
+#[derive(Copy, Clone, Debug, PartialEq)]
```

Comment:
@camshaft you mentioned you're not a fan of Copy in https://github.com/aws/s2n-tls/pull/4764#discussion_r1778907751. Do you think we can get away with it here, since it's more for reading than writing, or should I still avoid it? Things work fine with clone, the implementation is just slightly more annoying.

Reply:
Copy is usually fine, it just means you need to keep in mind how you end up using it (I think shared an example issue that can arise in the other comment)

Reply:
This type should be ok though